### PR TITLE
feat(auth): add refreshing tokens flow

### PR DESCRIPTION
**Pull request by ice-tychon**
### What does this PR do?
This PR adds a feature to refresh tokens, so a user is not logged out after their token expires.

### Changes brought by this PR
- The main change is `AuthInterceptor `, it catches 401 errors from BE, blocks other requests (because it's a `QueuedInterceptor`) and tries to refresh token. After that it sets a new token header.

- Added `DelegatedLoginService` with its dependencies to ion_identity_client. This thing makes the request to get a new token.
- Updated ion_identity_client/example a little bit.
- Added `AppLifecycleProvider`, it holds the `AppLifecycleState`

### CODE REVIEW

File: lib/app/features/core/providers/app_lifecycle_provider.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
```
Comment (ice-alcides) [inline lib/app/features/core/providers/app_lifecycle_provider.dart:3]: Providers should ideally not have UI imports. On the other hand, `AppLifecycleState` is just an enum. Let's decide if that's okay for us

File: lib/app/features/auth/services/refresh_tokens_service.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      _scheduleTokenRefresh(newToken);
+    } catch (e) {
+      // Retry after a short delay
+      Timer(const Duration(minutes: 1), () => _refreshToken(token));
+    }
+  }
+
+  void dispose() {
+    for (final timer in _refreshTimers.values) {
+      timer.cancel();
```
Comment (ice-alcides) [inline lib/app/features/auth/services/refresh_tokens_service.dart:68]: need to cheks if the `timer` is not null

File: lib/app/features/core/providers/app_lifecycle_provider.dart (hunk 1 above)
Comment (ice-tychon) [reply lib/app/features/core/providers/app_lifecycle_provider.dart:3]: Discussed.

**Review by ice-alcides — DISMISSED** [review]

File: packages/ion_identity_client/lib/src/auth/services/register_service.dart [hunk 3]
```diff
@@ -45,10 +45,12 @@ class RegisterService {
...
         )
         .flatMap(
-          (r) => tokenStorage.setToken(
-            username: username,
-            newToken: r.authentication.token,
-            onError: UnknownRegisterUserFailure.new,
+          (r) => TaskEither.tryCatch(
+            () => tokenStorage.setTokens(
+              username: username,
+              newTokens: r.authentication,
```
[the remark this answers is not in the document]
Comment (ice-tychon) [reply packages/ion_identity_client/lib/src/auth/services/register_service.dart:51]: Tested, now it works.

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**